homeassistant/components/qwikswitch.py

```diff
@@ -0,0 +1,180 @@
+"""
+Support for Qwikswitch lights and switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/qwikswitch
+"""
+
+import logging
+
+REQUIREMENTS = ['https://github.com/kellerza/pyqwikswitch/archive/v0.1.zip'
+                '#pyqwikswitch==0.1']
+DEPENDENCIES = []
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'qwikswitch'
+DISCOVER_LIGHTS = 'qwikswitch.light'
+DISCOVER_SWITCHES = 'qwikswitch.switch'
+
+ADD_DEVICES = None
+
+
+class QSToggleEntity(object):
+    # pylint: disable=line-too-long
+    """Representation of a Qwikswitch Entiry.
+
+    Implement base QS methods. Modeled around HA ToggleEntity[1] & should only
+    be used in a class that extends both QSToggleEntity *and* ToggleEntity.
+
+    Implemented:
+     - QSLight extends QSToggleEntity and Light[2] (ToggleEntity[1])
+     - QSSwitch extends QSToggleEntity and SwitchDevice[3] (ToggleEntity[1])
+
+    [1] https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py  # NOQA
+    [2] https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/light/__init__.py  # NOQA
+    [3] https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/switch/__init__.py  # NOQA
+    """
+
+    def __init__(self, qsitem, qsusb):
+        """Initialize the light."""
+        self._id = qsitem['id']
+        self._name = qsitem['name']
+        self._qsusb = qsusb
+        self._value = qsitem.get('value', 0)
+        self._dim = qsitem['type'] == 'dim'
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..100."""
+        return self._value if self._dim else None
+
+    # pylint: disable=no-self-use
+    @property
+    def should_poll(self):
+        """State Polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Check if On (non-zero)."""
+        return self._value > 0
+
+    def update_value(self, value):
+        """Decode QSUSB value & update HA state."""
+        self._value = value
+        # pylint: disable=no-member
+        super().update_ha_state()  # Part of Entity/ToggleEntity
+        return self._value
+
+    # pylint: disable=unused-argument
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        from homeassistant.components.light import ATTR_BRIGHTNESS
```

Comment:
This created an import loop with the `light/__init__.py` that imported `from homeassistant.component.qwikswitch`.

Reading up on python import loops, it can apparently be fixed by importing the complete module `import components.light` instead, but with PR2012 I can also fix it